### PR TITLE
DDO-852 WM Refactor

### PR DESCRIPTION
**Pull request by gmalkov**
Updates Workspace Manager chart as follows:
- Readiness/liveness probes
- Resource requests/limits
- Prometheus JVM monitoring
- ConfigMap hash annotations so that deployment is restarted on configMap changes
- Various syntax/formatting tweaks

The [WM in my environment](https://ap-argocd.dsp-devops.broadinstitute.org/applications/workspacemanager-gmalkov) is deployed from this PR, and is [sending Prometheus metrics to GCP monitoring](https://console.cloud.google.com/monitoring/metrics-explorer?project=terra-kernel-k8s&pageState=%7B%22xyChart%22:%7B%22dataSets%22:%5B%7B%22timeSeriesFilter%22:%7B%22filter%22:%22metric.type%3D%5C%22external.googleapis.com%2Fprometheus%2Fjvm_memory_bytes_used%5C%22%20resource.type%3D%5C%22k8s_container%5C%22%22,%22minAlignmentPeriod%22:%2260s%22,%22aggregations%22:%5B%7B%22perSeriesAligner%22:%22ALIGN_MEAN%22,%22crossSeriesReducer%22:%22REDUCE_SUM%22,%22groupByFields%22:%5B%22resource.label.%5C%22namespace_name%5C%22%22,%22resource.label.%5C%22pod_name%5C%22%22%5D%7D,%7B%22crossSeriesReducer%22:%22REDUCE_NONE%22%7D%5D%7D,%22targetAxis%22:%22Y1%22,%22plotType%22:%22LINE%22%7D%5D,%22options%22:%7B%22mode%22:%22COLOR%22%7D,%22constantLines%22:%5B%5D,%22timeshiftDuration%22:%220s%22,%22y1Axis%22:%7B%22label%22:%22y1Axis%22,%22scale%22:%22LINEAR%22%7D%7D,%22isAutoRefresh%22:true,%22timeSelection%22:%7B%22timeRange%22:%221h%22%7D%7D).

### CODE REVIEW

File: charts/workspacemanager/templates/deployment.yaml [hunk 1]
```diff
@@ -1,38 +1,104 @@
...
+      annotations:
+        {{- /* Automatically restart deployments on config map changes: */}}
+        checksum/{{ .Values.name }}-proxy-configmap: {{ include (print $.Template.BasePath "/configmaps/proxy-apache-httpd-configmap.yaml") . | sha256sum }}
+        checksum/{{ .Values.name }}-tcell-configmap: {{ include (print $.Template.BasePath "/configmaps/tcell_configmap.yaml") . | sha256sum }}
+        {{- if .Values.postgres.enabled }}
+        checksum/{{ .Values.name }}-postgres-initdb: {{ include (print $.Template.BasePath "/configmaps/postgres-init-db-configmap.yaml") . | sha256sum }}
+        {{- end }}
+        {{- if .Values.prometheus.enabled }}
+        checksum/{{ .Values.name }}-prometheus-configmap: {{ include (print $.Template.BasePath "/configmaps/prometheus-configmap.yaml") . | sha256sum }}
+        {{- end }}
```
Comment (wchamber) [inline charts/workspacemanager/templates/deployment.yaml:37]: Could you say more about how this works?
I'm would be very worried about allowing far off config changes to cause WM to rolling restart. Config updates should happen very intentionally per service. If this is referring to a config map that ends up being defined in another "unit of deployment" (today or next quarter), you start to get very mysterious action at a distance, and it's distressing how often config changes bring down services.
I'm ok with this if the restart for deployments are only for changes to the values of the workspace manager helm chart/helm files (and it'd be great if that was enforceable).

Comment (gmalkov) [reply charts/workspacemanager/templates/deployment.yaml:37]: > Could you say more about how this works?
Sure, I stole it from [here](https://sanderknape.com/2019/03/kubernetes-helm-configmaps-changes-deployments/). Basically it ensures that whenever one of these configmaps is changed as part of a Helm release/ArgoCD sync, the deployment restarts and picks up the changes.
> restart for deployments are only for changes to the values of the workspace manager helm chart/helm files
Yep, this is the case.

**Review by mflinn-broad — APPROVED** [review]
lgtm!

**Review by choover-broad — APPROVED** [review]
Nice! Looks great to me except for firewall rules -- WSM is private, so needs a CloudArmor policy.

PR for the CloudArmor policy is in progress here: https://broadworkbench.atlassian.net/browse/DDO-853

**Pull request merged**